substrate/frame/beefy/src/lib.rs

```diff
@@ -294,6 +302,54 @@ pub mod pallet {
 	}
 }
 
+#[cfg(any(feature = "try-runtime", test))]
+impl<T: Config> Pallet<T> {
+	/// Ensure the correctness of the state of this pallet.
+	///
+	/// This should be valid before or after each state transition of this pallet.
+	pub fn do_try_state() -> Result<(), sp_runtime::TryRuntimeError> {
+		Self::try_state_authorities()?;
+		Self::try_state_validators()?;
+
+		Ok(())
+	}
+
+	/// # Invariants
+	///
+	/// * `Authorities` should not exceed the `MaxAuthorities` capacity.
+	/// * `NextAuthorities` should not exceed the `MaxAuthorities` capacity.
+	fn try_state_authorities() -> Result<(), sp_runtime::TryRuntimeError> {
+		let authorities = <Authorities<T>>::get();
```

Comment:
A bounded vec can never have more elements than its maximum. Doing a `decode_len` here could detect an error, if there are more values in storage that would be truncated on the next read operation.

Reply:
I am now making use of `decode_len` now